src/main/java/fi/aalto/cs/apluscourses/presentation/module/ModuleListElementViewModel.java

```diff
@@ -0,0 +1,57 @@
+package fi.aalto.cs.apluscourses.presentation.module;
+
+import fi.aalto.cs.apluscourses.model.Module;
+import fi.aalto.cs.apluscourses.presentation.base.BaseViewModel;
+import fi.aalto.cs.apluscourses.presentation.base.ListElementViewModel;
+import java.awt.font.TextAttribute;
+import org.jetbrains.annotations.NotNull;
+
+public class ModuleListElementViewModel extends ListElementViewModel<Module> {
+
+  public ModuleListElementViewModel(@NotNull Module module) {
+    super(module);
+    module.stateChanged.addListener(this, BaseViewModel::onChanged);
+  }
+
+  public String getName() {
+    return getModel().getName();
+  }
+
+  public String getUrl() {
+    return getModel().getUrl().toString();
+  }
+
+  /**
+   * Returns a textual representation of the status of the module.
+   * @return A {@link String} describing the status.
+   */
+  public String getStatus() {
+    switch (getModel().stateMonitor.get()) {
+      case Module.NOT_INSTALLED:
+      case Module.FETCHED:
+        return "Not installed";
+      case Module.FETCHING:
+        return "Downloading...";
+      case Module.LOADING:
+        return "Installing...";
```

Comment:
Nit: could this be unified ("FETCHING -> "Fetching" or DOWNLOADING -> "Downloading")?

Reply:
I used term "fetch" instead of "download" because I thought it is more general term which can refer either downloading or file copying (like the current implementation) or whatever. The `Module` (base) class doesn't care about *how* its content is fetched, so it uses this general term.

However, at the level of UI, we know that we are doing downloading when we fetch (in the final product), so we show the user more specific description on what is done.